app/models/piece.rb

```diff
@@ -4,157 +4,188 @@ class Piece < ActiveRecord::Base
 
   # Validations
   validates :color, presence: true, inclusion: { in: ['white', 'black'] }
-  validates :x_pos, presence: true, numericality: { only_integer: true }, inclusion: { in: (0..7).to_a }
-  validates :y_pos, presence: true, numericality: { only_integer: true }, inclusion: { in: (0..7).to_a }
+  validates :x_pos, presence: true, numericality: { only_integer: true }, inclusion: { in: (0..8).to_a }
+  validates :y_pos, presence: true, numericality: { only_integer: true }, inclusion: { in: (0..8).to_a }
   validates :captured, inclusion: { in: [true, false] }
   validates :game, presence: true
 
   # Instance Methods
 
-  # This method can vary with different (inherited) pieces
-  # This method will change in future revisions
-  def valid_move_direction?(end_coord)
-    # Right now, only need to check if the move is horizontal,
-    # vertical, or diagonal. Other pieces (e.g. Horse) will
-    # contain varying logic.
-    begin
-      get_move_direction(end_coord)
-    rescue StandardError
-      return false
-    end
+  def move_to(end_coord)
+
+    #Check if move is vaild for Piece
+    return false unless self.valid_move?(end_coord)
+
+    capture?(end_coord)
+
+    #Update piece location and move count
+    self.update_attributes(:x_pos => end_coord[0], :y_pos => end_coord[1], :moves_made => self.moves_made + 1)
 
     true
+
   end
 
-  def get_move_direction(end_coord)
-    direction = case
-    when move_is_horizontal?(end_coord)
-      :horizontal
-    when move_is_vertical?(end_coord)
-      :vertical
-    when move_is_diagonal?(end_coord)
-      :diagonal
-    else
-      raise StandardError, "invalid move direction"
-    end
+  def capture(end_coord)
 
-    direction
-  end
+  #Capture logic
+   occuping_piece = game.pieces.where(x_pos: end_coord[0], y_pos: end_coord[1])
+
+   puts occuping_piece.inspect
 
-  # Takes a coordinate [x, y] and a direction :<direction>
-  def distance_from(end_coord, direction)
-    end_x_pos = end_coord[0]
-    end_y_pos = end_coord[1]
-
-    distance = case direction
-    when :horizontal
-      (end_x_pos - self.x_pos).abs
-    when :vertical
-      (end_y_pos - self.y_pos).abs
-    when :diagonal
-      (end_x_pos - self.x_pos).abs              # could have also used y distance, arbitrary choice
-    else
-      nil
+    if occuping_piece != nil
+      if self.color != occuping_piece.color
```

Comment:
This fails because of  
NoMethodError:
       undefined method `color' for #<Piece::ActiveRecord_AssociationRelation:0xbc8f9eb0>